pkg/kubelet/rkt/rkt.go

```diff
@@ -2034,7 +2034,7 @@ func (r *Runtime) AttachContainer(containerID kubecontainer.ContainerID, stdin i
 // Note: In rkt, the container ID is in the form of "UUID:appName", where UUID is
 // the rkt UUID, and appName is the container name.
 // TODO(yifan): If the rkt is using lkvm as the stage1 image, then this function will fail.
-func (r *Runtime) ExecInContainer(containerID kubecontainer.ContainerID, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, resize <-chan term.Size) error {
+func (r *Runtime) ExecInContainer(containerID kubecontainer.ContainerID, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool, resize <-chan term.Size, timeout time.Duration) error {
```

Comment:
For rkt it is possible to implement the timeout today -- refer to #27956, in particular https://github.com/kubernetes/kubernetes/pull/27956/commits/657cb46121c183e35bc0acc785e86fa201653724

Same for the nsenter backend: https://github.com/kubernetes/kubernetes/pull/27956/commits/d82bc859c8ffd612d0ec7c8eb4604d33f2502705